core/src/main/java/org/apache/accumulo/core/crypto/CryptoServiceFactory.java

```diff
@@ -56,11 +96,50 @@ public static CryptoService newInstance(AccumuloConfiguration conf, ClassloaderT
       throw new IllegalArgumentException();
     }
 
-    newCryptoService.init(conf.getAllPropertiesWithPrefix(Property.INSTANCE_CRYPTO_PREFIX));
-    return newCryptoService;
+    var encrypter = newCryptoService.getEncrypter();
+    encrypter.init(initParams);
+    return encrypter;
   }
 
-  public static CryptoService newDefaultInstance() {
-    return newInstance(DefaultConfiguration.getInstance(), ClassloaderType.JAVA);
+  public static List<CryptoService> getDecrypters(AccumuloConfiguration conf, ClassloaderType ct) {
+    String[] classes = conf.get(TABLE_CRYPTO_DECRYPT_SERVICES).split(",");
```

Comment:
Should this also load the TABLE_CRYPTO_ENCRYPT_SERVICE decrypter under the expectation that the service currently configured to encrypt will also be needed to decrypt? If not, should we add a note to the Property.java making it clear that the service much be set in both locations?

Reply:
That isn't a bad idea. I was going with the approach that if the decrypt service isn't in the config that we won't load it. Another option would be to make it flexible by adding the encrypt service to the list of decrypters if it isn't present.